### PR TITLE
WIP: GitHub pr comment template

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Simplify the GitHub PR comment from the CI.

## Why is it important?

Help with the debugging.

## Test

Steps failures

image

Aborted

image

## Issues

Requires https://github.com/elastic/apm-pipeline-library/pull/1078

### CODE REVIEW

File: .ci/github-comment.template [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  <% stepsErrors = stepsErrors << errorGitHub %>
+<%}%>
+<% if (stepsErrors?.size() != 0 && !statusSuccess) {%>
+### Steps errors [![${stepsErrors?.size()}](https://img.shields.io/badge/${stepsErrors?.size()}%20-red)](${jobUrl}/pipeline)
+<% if (stepsErrors?.size() > 10) {%>
+> Show only the first 10 steps failures
+<% }%>
+  <% stepsErrors?.takeRight(10)?.each{ c -> %>
+  <% description = (c?.displayDescription && c?.displayDescription != 'null') ? "<li>Description: <code>${c?.displayDescription}</code></l1>" : ''%>
+  <% url = (c?.url && c?.url != 'null') ? ". View more details on <a href=\"${c?.url}/?start=0\">here</a>" : ''%>
```
Comment (v1v) [inline .ci/github-comment.template:53]:
```suggestion
  <% url = (c?.url && c?.url != 'null') ? "View more details on <a href=\"${c?.url}/?start=0\">here</a>" : ''%>
```

**Comment by cachedout** [comment]
Copying [over feedback from closed issue](https://github.com/elastic/observability-robots/issues/501#issuecomment-848059031) via @jsoriano 

----

@v1v this is looking good, thanks!

I have some suggestions, mainly for the failed report, but they apply to some general parts. The first two things I'd check when a build fails are: the failing tests, and if I don't see anything relevant there, then I look for the logs of the step.
* For the failing tests I like how they are displayed in blue ocean in jenkins, and I think that if many tests fail, having them in a github comment can be difficult to follow, so I'd suggest:
  * If no junit reports are available, don't show the test errors section.
  * Show only up to a few (3?) test reports for builds where junit reports are available.
  * Reduce use of "expand to view..." (for example show directly the list of failed tests, and for each one have a single expandable view, with the message and the stacktrace).
  * "Test errors" and "genuine test errors" looks redundant to me, remove the genuine one.
  * Add a link to the blue ocean report with the junit results. If there are many failures I would prefer to see them there. 
* Links to failed steps logs can be removed if we can have the same links in the Github checks. 
* Show the flaky test report only when a flaky test was detected.
* More a matter of style: make it more compact (consider moving badges to the title, reduce title sizes).

Something like this (a bit broken, sorry):

----
<p>

### :broken_heart: Tests Failed [](https://apm-ci.elastic.co/blue/organizations/jenkins/apm-shared%2Fapm-pipeline-library-mbp%2FPR-1078/detail/PR-1078/18//tests) [Pipeline View](https://apm-ci.elastic.co/blue/organizations/jenkins/apm-shared%2Fapm-pipeline-library-mbp%2FPR-1078/detail/PR-1078/18//pipeline) [preview](http://apm-pipeline-library_1078.docs-preview.app.elstc.co/diff)


    
* `Test / test_assertion â€“ FailureTests` <details><summary>Expand to view the error details</summary><p>
* `Test / test_assertion â€“ FailureTests` <details><summary>Expand to view the error details</summary><p>
* `Test / test_assertion â€“ FailureTests` <details><summary>Expand to view the error details</summary><p>

<i><a href="">Link to jenkins report</a></i>


### Failed steps [](https://apm-ci.elastic.co/blue/organizations/jenkins/apm-shared%2Fapm-pipeline-library-mbp%2FPR-1078/detail/PR-1078/18//pipeline)
  
* `./mvnw clean test -V --batch-mode` . See logs <a href="https://apm-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/apm-shared/pipelines/apm-pipeline-library-mbp/pipelines/PR-1078/runs/18/steps/475/log/?start=0">here</a>

-----

If no junit and no flaky report is available:

----
<p>

### :broken_heart: Tests Failed [](https://apm-ci.elastic.co/blue/organizations/jenkins/apm-shared%2Fapm-pipeline-library-mbp%2FPR-1078/detail/PR-1078/18//tests) [Pipeline View](https://apm-ci.elastic.co/blue/organizations/jenkins/apm-shared%2Fapm-pipeline-library-mbp%2FPR-1078/detail/PR-1078/18//pipeline) [preview](http://apm-pipeline-library_1078.docs-preview.app.elstc.co/diff)

### Failed steps [](https://apm-ci.elastic.co/blue/organizations/jenkins/apm-shared%2Fapm-pipeline-library-mbp%2FPR-1078/detail/PR-1078/18//pipeline)
  
* `./mvnw clean test -V --batch-mode` . See logs <a href="https://apm-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/apm-shared/pipelines/apm-pipeline-library-mbp/pipelines/PR-1078/runs/18/steps/475/log/?start=0">here</a>

File: .ci/github-comment.template [hunk 2]
```diff
@@ -0,0 +1,46 @@
+<!-- See for further examaples https://github.com/elastic/apm-pipeline-library/blob/master/resources/github-comment-markdown.template -->
+
+<%if (statusSuccess) {%>
+## :green_heart: Build Succeeded
+<%} else if (buildStatus?.equals('ABORTED')) {%>
+## :grey_exclamation: Build Aborted
+  <%if (!build?.description != null && build?.description.toString().toLowerCase().contains('aborted')) {%>
+  > There is a new build on-going so the previous on-going builds have been aborted.
```
Comment (cachedout) [inline .ci/github-comment.template:8]:
```suggestion
  > There is an ongoing build so the previous ongoing builds have been aborted.
```

**Review by cachedout — APPROVED** [review]

**Pull request closed without merging**